// @angular-material-extensions/select-country/esm2015/lib/mat-select-country.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/mat-select-country.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { MatSelectCountryComponent } from './mat-select-country.component';
import { MatAutocompleteModule, MatButtonModule, MatIconModule, MatIconRegistry, MatInputModule, MatMenuModule } from '@angular/material';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';
import { DomSanitizer } from '@angular/platform-browser';
import { COUNTRIES_DB } from './db';
/**
 * @author Anthony Nahas
 * \@since 06.11.19
 */
export class MatSelectCountryModule {
    /**
     * @param {?} iconRegistry
     * @param {?} sanitizer
     */
    constructor(iconRegistry, sanitizer) {
        this.iconRegistry = iconRegistry;
        this.sanitizer = sanitizer;
        this.registerCountries();
    }
    /**
     * @return {?}
     */
    registerCountries() {
        for (const country of COUNTRIES_DB) {
            /** @type {?} */
            const countryAlpha2Code = country.alpha2Code.toLowerCase();
            this.iconRegistry
                .addSvgIcon(countryAlpha2Code, this.sanitizer
                .bypassSecurityTrustResourceUrl(`assets/svg-country-flags/svg/${countryAlpha2Code}.svg`));
        }
    }
}
MatSelectCountryModule.decorators = [
    { type: NgModule, args: [{
                declarations: [MatSelectCountryComponent],
                imports: [
                    CommonModule,
                    FormsModule,
                    ReactiveFormsModule,
                    HttpClientModule,
                    MatButtonModule,
                    MatMenuModule,
                    MatInputModule,
                    MatAutocompleteModule,
                    MatIconModule
                    // FlexLayoutModule
                ],
                exports: [MatSelectCountryComponent]
            },] }
];
/** @nocollapse */
MatSelectCountryModule.ctorParameters = () => [
    { type: MatIconRegistry },
    { type: DomSanitizer }
];
if (false) {
    /**
     * @type {?}
     * @private
     */
    MatSelectCountryModule.prototype.iconRegistry;
    /**
     * @type {?}
     * @private
     */
    MatSelectCountryModule.prototype.sanitizer;
}
