// @angular-material-extensions/select-country/esm5/lib/mat-select-country.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/mat-select-country.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { NgModule } from '@angular/core';
import { MatSelectCountryComponent } from './mat-select-country.component';
import { MatAutocompleteModule, MatButtonModule, MatIconModule, MatIconRegistry, MatInputModule, MatMenuModule } from '@angular/material';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { HttpClientModule } from '@angular/common/http';
import { DomSanitizer } from '@angular/platform-browser';
import { COUNTRIES_DB } from './db';
/**
 * @author Anthony Nahas
 * \@since 06.11.19
 */
var MatSelectCountryModule = /** @class */ (function () {
    function MatSelectCountryModule(iconRegistry, sanitizer) {
        this.iconRegistry = iconRegistry;
        this.sanitizer = sanitizer;
        this.registerCountries();
    }
    /**
     * @return {?}
     */
    MatSelectCountryModule.prototype.registerCountries = /**
     * @return {?}
     */
    function () {
        var e_1, _a;
        try {
            for (var COUNTRIES_DB_1 = tslib_1.__values(COUNTRIES_DB), COUNTRIES_DB_1_1 = COUNTRIES_DB_1.next(); !COUNTRIES_DB_1_1.done; COUNTRIES_DB_1_1 = COUNTRIES_DB_1.next()) {
                var country = COUNTRIES_DB_1_1.value;
                /** @type {?} */
                var countryAlpha2Code = country.alpha2Code.toLowerCase();
                this.iconRegistry
                    .addSvgIcon(countryAlpha2Code, this.sanitizer
                    .bypassSecurityTrustResourceUrl("assets/svg-country-flags/svg/" + countryAlpha2Code + ".svg"));
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (COUNTRIES_DB_1_1 && !COUNTRIES_DB_1_1.done && (_a = COUNTRIES_DB_1.return)) _a.call(COUNTRIES_DB_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
    };
    MatSelectCountryModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [MatSelectCountryComponent],
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        HttpClientModule,
                        MatButtonModule,
                        MatMenuModule,
                        MatInputModule,
                        MatAutocompleteModule,
                        MatIconModule
                        // FlexLayoutModule
                    ],
                    exports: [MatSelectCountryComponent]
                },] }
    ];
    /** @nocollapse */
    MatSelectCountryModule.ctorParameters = function () { return [
        { type: MatIconRegistry },
        { type: DomSanitizer }
    ]; };
    return MatSelectCountryModule;
}());
export { MatSelectCountryModule };
if (false) {
    /**
     * @type {?}
     * @private
     */
    MatSelectCountryModule.prototype.iconRegistry;
    /**
     * @type {?}
     * @private
     */
    MatSelectCountryModule.prototype.sanitizer;
}
